Lib/test/test_zipimport.py

```diff
@@ -669,6 +669,26 @@ def testBytesPath(self):
 class CompressedZipImportTestCase(UncompressedZipImportTestCase):
     compression = ZIP_DEFLATED
 
+    @support.cpython_only
+    def test_issue31602(self):
+        # There shouldn't be an assertion failure in zipimporter.get_source()
+        # in case of a bad zlib.decompress().
+        import zlib
```

Comment:
There is no guarantee the zlib module will be available, so at the top of the module you should do:
```python
try:
  import zlib
except ImportError:
  zlib = None
```
Then decorate this method with `unittest.skipIf(zlib is None, "zlib not available")`.

Reply:
Or use the `test.support.requires_zlib` decorator.

Reply:
doesn't this decorator guarantee that zlib will be available for the test?
```
@support.requires_zlib
class CompressedZipImportTestCase(UncompressedZipImportTestCase):
```

Reply:
Yes, it was just out of the visible context.

Reply:
Ah, then keep the decorator but still switch to the `try` block at the top for the import (importing in methods is icky).